lib/modules/datasource/go/index.ts

```diff
@@ -59,8 +59,9 @@ export class GoDatasource extends Datasource {
       return null;
     }
 
-    // ignore v0.0.0- pseudo versions that are used Go Modules - look up default branch instead
-    const tag = value && !value.startsWith('v0.0.0-2') ? value : undefined;
+    // ignore vX.Y.Z-(0.)? pseudo versions that are used Go Modules - look up default branch instead
+    const tag =
+      value && !/^v\d+\.\d+\.\d+-0?\.?.*/.test(value) ? value : undefined;
```

Comment:
😕 that regex is bad, it'll match any `1.2.3-...` like version

Reply:
I guess our tests missed it, I will revert

Reply:
why is this bad? if digest updates are not enabled it will not suggest updating it
the tests covered it